### PR TITLE
Use PSA_ERROR_INVALID_ARGUMENT for invalid cipher input sizes

### PR DESCRIPTION
**Pull request by frestr**
## Description
Use PSA_ERROR_INVALID_ARGUMENT for invalid cipher input sizes, as opposed to PSA_ERROR_BAD_STATE.

The spec on psa_cipher_finish() states that PSA_ERROR_INVALID_ARGUMENT should be returned when:

"The total input size passed to this operation is not valid for this particular algorithm. For example, the algorithm is a based on block cipher and requires a whole number of blocks, but the total input size is not a multiple of the block size."

In the current version, there is a distinction between encryption and decryption on whether INVALID_ARGUMENT or BAD_STATE is returned, but this is not a part of the spec.

## Status
**READY**

### CODE REVIEW

File: library/psa_crypto.c [hunk 1]
```diff
@@ -199,7 +199,7 @@ psa_status_t mbedtls_to_psa_error( int ret )
         case MBEDTLS_ERR_CIPHER_INVALID_PADDING:
```
Comment (ronald-cron-arm) [inline library/psa_crypto.c:199]: About the commit message: please add the reason of the change in the commit message. Something similar as what you wrote in the PR description https://github.com/ARMmbed/mbedtls/pull/3723#issue-494221290 will be perfect.

Note: Please avoid long lines in the commit message, a quite common policy being roughly less than 70 characters.

**Review by ronald-cron-arm — COMMENTED** [review]
Look good to me. I am just asking to explain the reason of the change in the commit message. The CI seems OK, just stuck for whatever reason.

File: library/psa_crypto.c (hunk 1 above)
Comment (frestr) [reply library/psa_crypto.c:199]: Thanks for the feedback. I've updated the commit message.

**Review by ronald-cron-arm — DISMISSED** [review]
Thanks for the nice commit message. LGTM.

**Review by yanesca — APPROVED** [review]
LGTM.

**Review by yanesca — CHANGES_REQUESTED** [review]
Oh, I am sorry I almost forgot: @frestr could you please add a ChangeLog entry?

**Comment by frestr** [comment]
@yanesca Sure, I added one now.

**Review by yanesca — CHANGES_REQUESTED** [review]
I apologise for the nitpick, but could we please mention in the Changelog entry that this is fixing a PSA API compliance issue?

**Comment by frestr** [comment]
No problem. I think it should be clearer now.

**Review by yanesca — CHANGES_REQUESTED** [review]
Thank you, looks perfect! One last thing: could you please add a `Signed-off-by` line to the commit message of the last commit? (Our automated DCO check fails without it.)

**Comment by frestr** [comment]
Right -- updated now.

**Review by yanesca — APPROVED** [review]
LGTM.

**Review by ronald-cron-arm — APPROVED** [review]

**Pull request merged**